test/integration/container/multimaster_test.go

```diff
@@ -284,86 +378,101 @@ func TestMultimasterSetup(t *testing.T) {
-	node3IP = sanitizeIP(run(t, "docker", "inspect", node_os+"-multimaster-node3", "--format='{{.NetworkSettings.IPAddress}}'"))
-
-	dirName := tempDir(t)
-	clusterYamlContent := fmt.Sprintf(clusterYAML, registryIP, registryPort)
-	clusterYaml := saveToFile(t, dirName, "cluster.yaml", clusterYamlContent)
-	fmt.Printf("clusterYAML file: %s\ncontents: \n%+v\n", clusterYaml, clusterYamlContent)
-
-	machinesYamlContent := fmt.Sprintf(string(machinesYAML), node0IP, node1IP, node2IP, node3IP)
-	machinesYaml := saveToFile(t, dirName, "machines.yaml", machinesYamlContent)
-	fmt.Printf("machinesYAML file: %s\ncontents: \n%+v\n", machinesYaml, machinesYamlContent)
-
-	_ = saveToFile(t, dirName, "repo-config.yaml", fmt.Sprintf(repoConfigMap, yumRepoIP))
-	_ = saveToFile(t, dirName, "docker-config.yaml", fmt.Sprintf(dockerConfigMap, registryIP, registryPort))
-
-	run(t, "../../../cmd/wksctl/wksctl", "apply",
-		fmt.Sprintf("--cluster=%s", clusterYaml), fmt.Sprintf("--machines=%s", machinesYaml),
-		fmt.Sprintf("--config-directory=%s", dirName),
-		"--verbose",
-		fmt.Sprintf("--controller-image=docker.io/weaveworks/wksctl-controller:%s", tag))
-
-	out := run(t, "../../../cmd/wksctl/wksctl", "kubeconfig",
-		fmt.Sprintf("--cluster=%s", clusterYaml), fmt.Sprintf("--machines=%s", machinesYaml))
-
-	var nodeList corev1.NodeList
-	for {
-		jsonOut, stderr := doRun("kubectl", "get", "nodes", "-o", "json", fmt.Sprintf("--kubeconfig=%s", kubeconfig(out)))
-		if stderr != "" {
-			log.Warnf("error from kubectl; ignoring: %s", stderr)
-			continue
-		}
-		if err := json.Unmarshal([]byte(jsonOut), &nodeList); err != nil {
-			log.Warnf("Error deserialising output of kubectl get nodes: %s", err)
-		}
-		log.Infof("The cluster currently has %d node(s)", len(nodeList.Items))
-		if len(nodeList.Items) == 4 {
-			break
-		}
-		time.Sleep(30 * time.Second)
-	}
-
-	assert.Len(t, nodeList.Items, 4)
-	assert.Len(t, nodes.Masters(nodeList).Items, 3)
-	assert.Len(t, nodes.Workers(nodeList).Items, 1)
 
-	expectedKubeletArgs := []string{"alsologtostderr=true", "container-runtime=docker"}
-	expectedApiServerArgs := []string{"alsologtostderr=true", "audit-log-maxsize=10000"}
+	tests := []struct {
+		name string
+		path string
+		skip bool
+	}{
+		{
+			name: CENTOS,
+			path: "../../../examples/footloose/centos7/docker/multimaster.yaml",
+			skip: false,
+		},
+		{
+			name: UBUNTU,
+			path: "../../../examples/footloose/ubuntu1804/docker/multimaster.yaml",
+			skip: true,
+		},
+	}
 
-	for i := 0; i < 4; i++ {
-		for _, kubeletArg := range expectedKubeletArgs {
-			log.Infof("Checking kubelet arg (%s) on node%d", kubeletArg, i)
-			run(t, "footloose",
-				"-c", "../../../examples/footloose/"+node_os+node_version+"/docker/multimaster.yaml",
-				"ssh", fmt.Sprintf("root@node%d", i), fmt.Sprintf("ps -ef | grep -v 'ps -ef' | grep /usr/bin/kubelet | grep %s", kubeletArg))
+	for _, tc := range tests {
+		tc := tc
```

Comment:
This seems like some leftover cruft...

Reply:
This is the section that actually runs the tests...